code/xcvm/lib/core/src/network.rs

```diff
@@ -5,17 +5,26 @@ use alloc::vec::Vec;
 use parity_scale_codec::{Decode, Encode};
 use scale_info::TypeInfo;
 use serde::{Deserialize, Serialize};
+use serde_with::{base64::Base64, serde_as};
 
 /// The interpreter origin, composite of a user origin and a salt.
-#[cfg_attr(feature = "std", derive(schemars::JsonSchema))]
```

Comment:
json schema is used to provide cw interface json schema for people to be generated. if this type is part of public contract interface, it should be json schema

Reply:
It’s implemented below on line 198.  The reason for lack of derive is because the derive(JsonSchema) doesn’t support serde_as.  I’m looking into some better ways of handling this without the need for all the explicit implementation.

Reply:
This turned out to be simpler than I thought. Done.